python/numpy.h

```diff
@@ -148,14 +144,15 @@ bool memory_overlaps(const py::array_t<T> &data, const View &view) {
   const auto begin = view.begin();
   const auto end = view.end();
   if (begin == end) {
+    // Scalars cannot overlap with numpy arrays in a way that matters to copy.
```

Comment:
Why does this comment refer to scalars? Scalars should not have `begin==end`?

Reply:
Hm, right. Seems I confused this with how `memory_bounds` used to work before the last fix.

But the implementation should still be correct for scalars. Removing the comment.

Reply:
Also, why is the check required at all? Isn't the condition below sufficient?

Reply:
Yes, I think so